### PR TITLE
Feature/scalable sample

### PR DESCRIPTION
**Pull request by atoennis**
## Description
The goal is to make the sample application more scalable while at the same time allowing it to be easy to clone the project and start with a blank slate.  The following actions were taken to help accomplish the goal:
* All code for the sample project as been moved to the `sample` source set
* The `main` source set has only a blank activity
* The `main` source set includes a barebone Dagger 2 implementation

To start a new project, clone the repo and delete the `sample` source set.  There might be some other small cleanup involved.  A future update to the readme can help outline the steps.

## Note
I had to update the excludes for FindBug since it was complaining about things in generated Dagger code.

### CODE REVIEW

**Comment by atoennis** [comment]
The next evolution of this would be to modularize the app.  This could also have the side benefit of working well with instant apps.

**Comment by weverb2** [comment]
You can run the main source set but It'll give you a blank activity.  You have to run the sample source set to get the sample app. - Adam Toennis

File: app/src/sample/java/com/mycompany/myapp/ui/main/SamplePresenter.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+import rx.Observable;
+import rx.Subscription;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
+
+public class SamplePresenter {
+    private static final String EXTRA_STATE = "MainPresenterState";
+
+    public interface MainViewContract {
```
Comment (weverb2) [inline app/src/sample/java/com/mycompany/myapp/ui/main/SamplePresenter.java]: Change to SampleViewContract

**Comment by atoennis** [comment]
@patrickhammond I'm with @stuartsoft reviewing past attempts at making code cleanup of the starter project easier and referenced this PR.  Do you recall why it wasn't pulled in?

**Comment by patrickhammond** [comment]
@atoennis I do not. Anything in past logged conversations from Aug where this might have come up?

**Pull request closed without merging**